### ISSUE TITLE
Getting all 0s as prediction

### ISSUE THREAD

**Issue by shubhamgoel27** [mattdutson/xview2#2]
Hi,

I was trying to replicate your code and use it for my research. I trained the model with the train.py script and the accuracy seemed to increase with increasing epochs (I trained for around 20). But I'm getting all 0s as prediction using the test.py script (probably the model is predicting the undamaged/unclassified class for all pixels. 

I'm not exactly sure how to debug this. 

Thanks.

**Comment by shubhamgoel27** [comment]
Also, I believe the training script does not provide an argument to make the localization model different. But the test script does have an option for specifying it separately. 

Is it to make the code workable with localization models trained by others, or am I missing something?

**Comment by mattdutson** [comment]
@shubhamgoel27 We had the exact same problem when working on this project. As far as I know, this is not due to a bug in the implementation, but rather arises from the highly unbalanced nature of the pixel classes (~97% of the labels are 0 in the training set).

The solution is to decrease the weight for class 0. This can be accomplished by modifying line 45 of `train.py`. Maybe try `0.01` or `0.03`. If you do this, the accuracy may decrease, but if you check the output you will start seeing the model predict something non-trivial. The decrease in accuracy is fine since what you really care about are precision/recall in classes 1-4.

If this doesn't work, I would also recommend lowering the learning rate (line 45 of `train.py`).

**Comment by mattdutson** [comment]
For your second question, the idea is to train two models using `train.py`: one which does well in localization and another which has high recall for the four damage classes (e.g. class weights `[0.0, 1.0, 3.0, 3.0, 1.0]`). The predictions of these two models are then merged.

**Comment by mattdutson** [comment]
Best of luck to you!

**Comment by shubhamgoel27** [comment]
Sure. I read the other issue and thought the class weights from the repo would be just fine for the training. I'll try modifying the weights and train it again. Thanks a lot for the prompt response.
